azavea/raster-foundry
# Drag and drop files into upload modal

This PR integrates the new upload design from https://github.com/azavea/raster-foundry/pull/68 and adds support for dragging and dropping files to the upload modal. 

To test: 
 * Add some files via the browse link. Add some more files. The new files should be prepended to the list. 
 * Click the red X to delete some files.
 * Drag some files onto the upload area. While dragging over the left side of the upload modal, a dashed border should appear. The file list should be updated on dropping the files.
 * Click continue to upload files. 
 * Close the modal and then reopen it. The file list should be clear.

Known issues:
 * Originally, I wanted to only add files to the list if they are new. This isn't possible, as far as I can tell, because there's no way to tell if two files are identical. The file object only contains the file name, and it's possible that two files can have the same name but different paths. However, the desired behavior (of only adding if new) does occur when the user attempts to add the same file twice in a row using the file chooser. In this case, the onchange event is not fired on the second attempt to add the file. This is somewhat problematic because it's inconsistent, but I don't think it will be a real problem because I doubt users will try to add the same file twice in a row.
 * When files are added to the list, an animation is supposed to run. I can't get it to work correctly, but will ask Design for assistance.

Connects #43
